### PR TITLE
Improved Zeta Function and Added Tests

### PR DESCRIPTION
**Pull request by ShikharJ**

### CODE REVIEW

**Comment by ShikharJ** [comment]
ping @isuruf @certik

**Comment by isuruf** [comment]
Can you avoid code duplication?

**Comment by ShikharJ** [comment]
'Zeta' is a 'TwoArg' function. To incorporate this into our implementation of 'LeviCivita' and 'FunctionSymbol' , we'll need to templatise that implementation and introduce a couple of 'if-else' statements at places. Should I go for that implementation?

File: symengine/derivative.cpp [hunk 1]
```diff
@@ -74,10 +74,39 @@ class DiffImplementation
...
+                if (eq(*a, *x)) {
+                    found_x = true;
+                    count++;
+                } else if (count < 2 and neq(*a->diff(x), *zero)) {
+                    count++;
+                }
+            }
+            if (count == 1 and found_x) {
+                return Derivative::create(self.rcp_from_this(), {x});
+            }
```
Comment (ShikharJ) [inline symengine/derivative.cpp]: Only the above lines after the `} else {` statement are same as our implementation for `LeviCivta` and `FunctionSymbol`. The rest has been modified to do cater to `Zeta`. Should I instead make a function for these lines?

**Comment by ShikharJ** [comment]
Ping @isuruf

File: symengine/derivative.cpp (hunk 1 above)
Comment (isuruf) [inline symengine/derivative.cpp]: Above lines after `} else {` can be replaced by,
```C++
if (eq(*diff, *zero)) {
    if (eq(*self.get_s(), *x)) {
        return Derivative::create(self.rcp_from_this(), {x});
    }
}
```

File: symengine/derivative.cpp [hunk 2]
```diff
@@ -74,10 +74,39 @@ class DiffImplementation
...
+            vec_basic v = self.get_args();
+            v[0] = get_dummy(self, "xi_1");
```
Comment (isuruf) [inline symengine/derivative.cpp]: You are not using `v`, so can you assign this to a variable called `dummy` and use it?

File: symengine/derivative.cpp [hunk 3]
```diff
@@ -74,10 +74,39 @@ class DiffImplementation
...
+            map_basic_basic m;
+            insert(m, v[0], self.get_args()[0]);
```
Comment (isuruf) [inline symengine/derivative.cpp]: Use `self.get_s()` instead of `self.get_args()[0]`

File: symengine/derivative.cpp [hunk 4]
```diff
@@ -74,10 +74,39 @@ class DiffImplementation
...
+            diff = add(
+                diff,
+                mul(m1, make_rcp<const Subs>(
+                            Derivative::create(
+                                self.create(v[0], self.get_args()[1]), {v[0]}),
```
Comment (isuruf) [inline symengine/derivative.cpp]: Use `self.get_a()` instead of `self.get_args()[1]`

**Comment by ShikharJ** [comment]
Updated @isuruf.

**Pull request merged**